tests/end2endtest/test_indications.py

```diff
@@ -0,0 +1,568 @@
+"""
+End2end tests of indications from a wbem server. This tests:
+1. creating the subscription in the container
+2. Creating the pywbemlistener
+3. Receiving indications from the server. It uses only OpenPegasus WBEM server
+because OpenPegasus defines a provider in its test environment specifically to
+send indications based on a method invocation and we have a container
+implementation of OpenPegasus.
+
+"""
+
+from __future__ import absolute_import, print_function
+
+import socket
+import time
+import sys
+import os
+import datetime
+import logging
+
+from .utils.pytest_extensions import skip_if_unsupported_capability
+
+# Note: The wbem_connection fixture uses the es_server fixture, and
+# due to the way py.test searches for fixtures, it also must be imported.
+# pylint: disable=unused-import,wrong-import-order, relative-beyond-top-level
+from .utils.pytest_extensions import wbem_connection  # noqa: F401
+from pytest_easy_server import es_server  # noqa: F401
+from .utils.pytest_extensions import default_namespace  # noqa: F401
+# pylint: enable=unused-import,wrong-import-order, relative-beyond-top-level
+
+# pylint: disable=wrong-import-position, wrong-import-order, invalid-name
+# pylint: disable=relative-beyond-top-level, disable=redefined-builtin
+from ..utils import import_installed
+
+pywbem = import_installed('pywbem')
+from pywbem import WBEMServer, WBEMListener, CIMClassName, \
+    Error, Uint32, WBEMSubscriptionManager, \
+    configure_logger  # noqa: E402
+# pylint: enable=relative-beyond-top-level, enable=redefined-buitin
+# pylint: enable=wrong-import-position, wrong-import-order, invalid-name
+
+
+def test_indications(wbem_connection):  # pylint: disable=redefined-outer-name
+    """
+    Test indication subscription to a container and reception of indications
+    from the container.
+    """
+    skip_if_unsupported_capability(wbem_connection, 'interop')
+
+    # Get a real IP address for the machine to insert into  the
+    # indication_destination_host variable. This must be an address available
+    # to both the test machine/vm/etc. and the container in which OpenPegasus
+    # is running so OpenPegasus can route indications to the listener.
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.connect(("8.8.8.8", 80))
+    indication_dest_host = s.getsockname()[0]
+
+    # Use the empty string host that allows receving indications addressed to
+    # any valid valid destination on the system.
+    listener_host = ""
+    port = 5000
+    log = True
+    verbose = True
+
+    # define repeat_loop and indication_count variables to rerun the test with
+    # each of these pairs. (repeat_loop, indication_count)
+    # FUTURE: Expand this to more indications when we resolve issue #3022,
+    # the container not returning all of the indications.  This appears to
+    # be much more consistent with python 3.7 and 3.7 PACKAGE_LEVEL=minimum
+    if sys.version_info[1] == 7:   # Use minimum send ct while issue #3022
```

Comment:
Can you please be specific about the major version? If you really mean this case to be taken for 2.7 and 3.7, then say:
```
if sys.version_info[0:2] in ((2, 7), (3, 7)):
```

Second: I do not understand how this version selection matches the comment about 3.7

Reply:
Fixed.